deepmedic/frontEnd/trainSession.py

```diff
@@ -19,148 +19,186 @@
-                                    num_classes = model_params.numberClasses,
-                                    model_name = model_params.cnnModelName,
-                                    cfg = self._cfg )
-        
+                                    log=self._log,
+                                    mainOutputAbsFolder=self._main_out_folder_abs,
+                                    folderForSessionCnnModels=self._out_folder_models,
+                                    folderForPredictionsVal=self._out_folder_preds,
+                                    folderForFeaturesVal=self._out_folder_fms,
+                                    num_classes=model_params.numberClasses,
+                                    model_name=model_params.cnnModelName,
+                                    cfg=self._cfg)
+
         self._log.print3("")
         self._log.print3("=============   NEW TRAINING SESSION     ==============\n")
         self._params.print_params()
         self._log.print3("=======================================================\n")
-        
+
         return self._params
-    
-    
+
     def run_session(self, *args):
         (sess_device,
          model_params,
          reset_trainer) = args
-        
+
         graphTf = tf.Graph()
-        
+
         with graphTf.as_default():
-            with graphTf.device(sess_device): # Explicit device assignment, throws an error if GPU is specified but not available.
+            # Explicit device assignment, throws an error if GPU is specified but not available.
+            with graphTf.device(sess_device):
                 self._log.print3("=========== Making the CNN graph... ===============")
                 cnn3d = Cnn3d()
                 with tf.variable_scope("net"):
-                    cnn3d.make_cnn_model( *model_params.get_args_for_arch() )
+                    cnn3d.make_cnn_model(*model_params.get_args_for_arch())
                     # I have now created the CNN graph. But not yet the Optimizer's graph.
-            
-            # No explicit device assignment for the rest. Because trained has piecewise_constant that is only on cpu, and so is saver.        
+
+            # No explicit device assignment for the rest.
+            # Because trained has piecewise_constant that is only on cpu, and so is saver.
             with tf.variable_scope("trainer"):
                 self._log.print3("=========== Building Trainer ===========\n")
-                trainer = Trainer( *( self._params.get_args_for_trainer() + [cnn3d] ) )
-                trainer.create_optimizer( *self._params.get_args_for_optimizer() ) # Trainer and net connect here.
-                
+                trainer = Trainer(*(self._params.get_args_for_trainer() + [cnn3d]))
+                trainer.create_optimizer(*self._params.get_args_for_optimizer())  # Trainer and net connect here.
+
+                tensorboard_loggers = self.create_tensorboard_loggers(['train', 'val'],
```

Comment:
Does not need an initializer, nor a saver.

I would definitely not put it in a "net" or "trainer" scope. Lets keep those clean for very specific variables.

It can just come out of scope (one tab in the left), or if we want to be more explicit, lets make an "others" scope:
```suggestion
            with tf.variable_scope("others"):
                tensorboard_loggers = self.create_tensorboard_loggers(['train', 'val'],
```